src/app/shared/services/HttpService/HttpService.ts

```diff
@@ -0,0 +1,29 @@
+import VueRouter from 'vue-router';
+import axios, { AxiosError, AxiosInstance, AxiosRequestConfig, AxiosResponse } from 'axios';
+import { Store } from 'vuex';
+import { IState } from '../../../state';
+import { setupResponseInterceptor } from './setupResponseInterceptor';
+import { setupRequestInterceptor } from './setupRequestInterceptor';
+
+export interface IHttpService extends AxiosInstance {
+  store?: Store<IState>;
+  router?: VueRouter;
+}
+
+let baseURL;
+
+/* istanbul ignore if */
+if (!CLIENT) {
```

Comment:
I think this should be removed.

Reply:
it throws an error when it is loaded on the clientside. 
`fs` is not defined.

Reply:
I mean to remove this whole block ```  if (!CLIENT) {
  const config = require('../../../config/AppConfig');
  baseURL = config.api.baseUrl;
}```

we should load the baseUrl from the store (state.app.config)

Reply:
I see, at this stage, the `state` is not accessible.
I can think of passing the state as a param to the `httpService` to solve this, something like that:

```
function prepareRequest(state){
const HttpService: IHttpService = axios.create({
  baseURL: state.app.config.baseUrl,
});

setupRequestInterceptor(HttpService);
setupResponseInterceptor(HttpService);

return HttpService
}
```
However, this will require several changes, We can open a new issue, or we can revert back the changes related to handling the external URL in the request interceptor. what do you think? - I'm fine with both.

Reply:
let's keep your changes for now and revisit later 👍